src/v/archival/ntp_archiver_service.cc

```diff
@@ -1865,8 +1865,9 @@ ss::future<> ntp_archiver::apply_archive_retention() {
     }
 
     if (
-      res.value().offset
-      == _manifest_view->stm_manifest().get_archive_start_offset()) {
+      res.value().offset == model::offset{}
+      || res.value().offset
```

Comment:
What's the proper channel to get back to the archive start offset getting back to `model::offset{}`?

Reply:
My stab at an answer, but happy to be pointed to the right answer if there is one https://github.com/redpanda-data/redpanda/pull/11772

Reply:
Hmm. My mental model is that there's two ways to model the archive being empty:
1. archive start offset is `model::offset{}`
2. archive start offset is equal to stm region start offset

`ntp_archiver::stm_retention_needed` doesn't agree with my mental model though. I think we should just update
it to match the two cases above. @Lazin what do you think?

Reply:
We should probably add a test for this too (i.e. retention needs to remove the entire archive and some segments from the stm region).

Reply:
the both approaches are used. first, before first spillover manifest is created the archive start offset is model::offset{}. Then, when the retention removes everything it's equal to stm start offset.

Reply:
Right, but `ntp_archiver::stm_retention_needed` is only aware of the first approach. So I guess we should just fix that.